modules/dnn/test/test_model.cpp

```diff
@@ -0,0 +1,182 @@
+    std::vector<Rect2d> refBoxes = {Rect2d(240.032, 52.994, 135.408, 72.134),
+                                    Rect2d(112.320, 109.865, 192.067, 200.220),
+                                    Rect2d(57.658, 140.608, 117.603, 249.683)};
+
+    std::string img_path = _tf("dog416.png");
+    std::string weights_file = _tf("yolo-voc.weights");
+    std::string config_file = _tf("yolo-voc.cfg");
+
+    float scale = 1.0 / 255.0;
+    Size size{416, 416};
+    bool swapRB = true;
+
+    double confThreshold = 0.24;
+    double scoreDiff = (target == DNN_TARGET_OPENCL_FP16 || target == DNN_TARGET_MYRIAD) ? 1e-2 : 8e-5;
+    double iouDiff = 0.014;
+    double nmsThreshold = (target == DNN_TARGET_MYRIAD) ? 0.397 : 0.4;
+
+    testDetectModel(weights_file, config_file, img_path, refClassIds, refConfidences,
+                    refBoxes, scoreDiff, iouDiff, confThreshold, nmsThreshold, size,
+                    Scalar(), scale, swapRB);
+}
+
+TEST_P(Test_Model, DetectionOutput)
+{
+    std::vector<int> refClassIds = {7, 12};
+    std::vector<float> refConfidences = {0.991359f, 0.94786f};
+    std::vector<Rect2d> refBoxes = {Rect2d(491.822, 81.1668, 211.751, 98.0672),
+                                    Rect2d(132.093, 223.903, 206.984, 343.257)};
+
+    std::string img_path = _tf("dog416.png");
+    std::string weights_file = _tf("resnet50_rfcn_final.caffemodel");
+    std::string config_file = _tf("rfcn_pascal_voc_resnet50.prototxt");
+
+    Scalar mean = Scalar(102.9801, 115.9465, 122.7717);
+    Size size{800, 600};
+
+    double scoreDiff = (backend == DNN_BACKEND_OPENCV && target == DNN_TARGET_OPENCL_FP16) ?
+                        4e-3 : default_l1;
+    double iouDiff = 0.016;
+    float confThreshold = 0.8;
+    float nmsThreshold = 0;
+
+    testDetectModel(weights_file, config_file, img_path, refClassIds, refConfidences, refBoxes,
+                    scoreDiff, iouDiff, confThreshold, nmsThreshold, size, mean);
+}
+
+
+TEST_P(Test_Model, DetectionMobilenetSSD)
+{
+    Mat ref = blobFromNPY(_tf("mobilenet_ssd_caffe_out.npy"));
+    ref = ref.reshape(1, ref.size[2]);
+
+    std::string img_path = _tf("street.png");
+    Mat frame = imread(img_path);
+    int frameWidth  = frame.cols;
+    int frameHeight = frame.rows;
+
+    std::vector<int> refClassIds;
+    std::vector<float> refConfidences;
+    std::vector<Rect2d> refBoxes;
+    for (int i = 0; i < ref.rows; i++)
+    {
+        refClassIds.emplace_back(ref.at<float>(i, 1));
+        refConfidences.emplace_back(ref.at<float>(i, 2));
+        int left   = ref.at<float>(i, 3) * (frameWidth - 1);
+        int top    = ref.at<float>(i, 4) * (frameHeight - 1);
+        int right  = ref.at<float>(i, 5) * (frameWidth - 1);
+        int bottom = ref.at<float>(i, 6) * (frameHeight - 1);
+        int width  = right  - left + 1;
+        int height = bottom - top + 1;
+        refBoxes.emplace_back(left, top, width, height);
+    }
```

Comment:
Please fix the rest of tests correspondingly. Perhaps, you need to write a separate method to extract boxes from reference blob.